cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -0,0 +1,399 @@
+        self.out_map['RGW_ZONE_NAME'] = self._arg_parser.rgw_zone_name
+
+    def gen_shell_out(self):
+        self._gen_output_map()
+        shOutIO = StringIO()
+        for k, v in self.out_map.items():
+            if v:
+                shOutIO.write('export {}={}{}'.format(k, v, LINESEP))
+        shOut = shOutIO.getvalue()
+        shOutIO.close()
+        return shOut
+
+    def gen_json_out(self):
+        self._gen_output_map()
+        json_out = [
+            {
+                "name": "rook-ceph-mon-endpoints",
+                "kind": "ConfigMap",
+                "data": {
+                    "data": self.out_map['ROOK_EXTERNAL_CEPH_MON_DATA'],
+                    "maxMonId": "0",
+                    "mapping": {}
+                }
+            },
+            {
+                "name": "rook-ceph-mon",
+                "kind": "Secret",
+                "data": {
+                    "admin-secret": 'admin-secret',
+                    # TODO: add the argument --cluster-name = default: openshift-storage
+                    "cluster-name": self.out_map['CLUSTER_NAME'],
+                    "fsid": self.out_map['ROOK_EXTERNAL_FSID'],
+                    "mon-secret": "mon-secret"
+                },
+            },
+            {
+                "name": "rook-ceph-operator-creds",
+                "kind": "Secret",
+                "data": {
+                    "userID": self.out_map['ROOK_EXTERNAL_USERNAME'],
+                    "userKey": self.out_map['ROOK_EXTERNAL_USER_SECRET']
+                }
+            },
+            {
+                "name": "rook-csi-rbd-node",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-rbd-node',
+                    "userKey": self.out_map['CSI_RBD_NODE_SECRET_SECRET']
+                }
+            },
+            {
+                "name": "rook-csi-rbd-provisioner",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-rbd-provisioner',
+                    "userKey": self.out_map['CSI_RBD_PROVISIONER_SECRET']
+                },
+                "pool-name": self.out_map['RBD_POOL_NAME']
+            },
+            {
+                "name": "rook-csi-cephfs-node",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-cephfs-node',
+                    "userKey": self.out_map['CSI_CEPHFS_NODE_SECRET']
+                }
+            },
+            {
+                "name": "rook-csi-cephfs-provisioner",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-cephfs-provisioner',
+                    "userKey": self.out_map['CSI_CEPHFS_PROVISIONER_SECRET']
+                },
+                "filesystem-name": self.out_map['CEPHFS_FS_NAME'],
+                "pool-name": self.out_map['CEPHFS_POOL_NAME']
+            },
```

Comment:
We need a new section:

```
            {
                "name": "rook-ceph-retain-bucket",
                "kind": "StorageClass",
                "data": {
                    "endpoint": self.out_map['RGW_GATEWAY_ENDPOINT']
                },
            }
```

Reply:
Adderssed...
PS: changed the `RGW_GATEWAY_ENDPOINT` to `RGW_ENDPOINT`